pkg/virt-handler/vm.go

```diff
@@ -3005,173 +3000,179 @@ func (c *VirtualMachineController) vmUpdateHelperDefault(origVMI *v1.VirtualMach
-				return fmt.Errorf("failed to set up file ownership for /dev/vhost-vsock: %v", err)
-			}
+	return errors.NewAggregate(errorTolerantFeaturesError)
+}
+
+func (c *VirtualMachineController) getPreallocatedVolumes(vmi *v1.VirtualMachineInstance) []string {
+	var preallocatedVolumes []string
+	for _, volumeStatus := range vmi.Status.VolumeStatus {
+		if volumeStatus.PersistentVolumeClaimInfo != nil && volumeStatus.PersistentVolumeClaimInfo.Preallocated {
+			preallocatedVolumes = append(preallocatedVolumes, volumeStatus.Name)
 		}
+	}
+	return preallocatedVolumes
+}
 
-		lessPVCSpaceToleration := c.clusterConfig.GetLessPVCSpaceToleration()
-		minimumPVCReserveBytes := c.clusterConfig.GetMinimumReservePVCBytes()
+func (c *VirtualMachineController) handleRunningVMI(vmi *v1.VirtualMachineInstance, cgroupManager cgroup.Manager) error {
+	if err := c.hotplugSriovInterfaces(vmi); err != nil {
+		log.Log.Object(vmi).Error(err.Error())
+	}
 
-		// initialize disks images for empty PVC
-		hostDiskCreator := hostdisk.NewHostDiskCreator(c.recorder, lessPVCSpaceToleration, minimumPVCReserveBytes, virtLauncherRootMount)
-		err = hostDiskCreator.Create(vmi)
-		if err != nil {
-			return fmt.Errorf("preparing host-disks failed: %v", err)
-		}
+	if err := c.hotplugVolumeMounter.Mount(vmi, cgroupManager); err != nil {
+		return err
+	}
 
-		if virtutil.IsSEVVMI(vmi) {
-			sevDevice, err := safepath.JoinNoFollow(virtLauncherRootMount, filepath.Join("dev", "sev"))
-			if err != nil {
-				return err
-			}
-			if err := diskutils.DefaultOwnershipManager.SetFileOwnership(sevDevice); err != nil {
-				return fmt.Errorf("failed to set SEV device owner: %v", err)
-			}
-		}
+	if err := c.getMemoryDump(vmi); err != nil {
+		return err
+	}
 
-		if virtutil.IsNonRootVMI(vmi) {
-			if err := c.nonRootSetup(origVMI); err != nil {
-				return err
-			}
-		}
-		for _, fs := range vmi.Spec.Domain.Devices.Filesystems {
-			socketPath, err := isolation.SafeJoin(isolationRes, virtiofs.VirtioFSSocketPath(fs.Name))
-			if err != nil {
-				return err
-			}
-			if err := diskutils.DefaultOwnershipManager.SetFileOwnership(socketPath); err != nil {
-				return err
-			}
-		}
+	isolationRes, err := c.podIsolationDetector.Detect(vmi)
+	if err != nil {
+		return fmt.Errorf(failedDetectIsolationFmt, err)
+	}
 
-		// set runtime limits as needed
-		err = c.podIsolationDetector.AdjustResources(vmi, c.clusterConfig.GetConfig().AdditionalGuestMemoryOverheadRatio)
+	if err := c.downwardMetricsManager.StartServer(vmi, isolationRes.Pid()); err != nil {
+		return err
+	}
+
+	if err := c.setupNetwork(vmi, netsetup.FilterNetsForLiveUpdate(vmi)); err != nil {
+		log.Log.Object(vmi).Error(err.Error())
+		c.recorder.Event(vmi, k8sv1.EventTypeWarning, "NicHotplug", err.Error())
+	}
+
+	return nil
+}
+
+func (c *VirtualMachineController) handleStartingVMI(vmi *v1.VirtualMachineInstance, cgroupManager cgroup.Manager, disksInfo *map[string]*containerdisk.DiskInfo) (bool, error) {
```

Comment:
Maps are reference types:
```suggestion
func (c *VirtualMachineController) handleStartingVMI(vmi *v1.VirtualMachineInstance, cgroupManager cgroup.Manager, disksInfo map[string]*containerdisk.DiskInfo) (bool, error) {
```